### PR TITLE
Move process logger getter and setter to the Runtime static class

### PR DESCRIPTION
**Pull request by pepone**
With this small PR getter and setter of the process logger are now static methods of the communicator, place in `Communicator-PocessLogger.cs` that has a new Communicator class part.

The `StringVersion()` and `IntVersion()` Util methods are replaced by matching static properties on the Communicator.

The Util class will removed in a follow up PR whe we get rid of the `TransportPluginFacade`

### CODE REVIEW

File: csharp/src/Ice/Communicator.cs [hunk 1]
```diff
@@ -144,6 +144,17 @@ private class ObserverUpdater : Instrumentation.IObserverUpdater
...
+        /// <summary>Returns the Ice version in the form A.B.C, where A indicates the major version, B indicates the
+        /// minor version, and C indicates the patch level.</summary>
+        /// <returns>The Ice version.</returns>
+        public static string StringVersion => "4.0.0-alpha.0"; // "A.B.C", with A=major, B=minor, C=patch
+
+        /// <summary>Returns the Ice version as an integer in the form A.BB.CC, where A indicates the major version,
+        /// BB indicates the minor version, and CC indicates the patch level. For example, for Ice 3.3.1, the returned
+        /// value is 30301.</summary>
+        /// <returns>The Ice version.</returns>
+        public static int IntVersion => 40000; // AABBCC, with AA=major, BB=minor, CC=patch
```
Comment (externl) [inline csharp/src/Ice/Communicator.cs:156]: Is Communicator really the best place for the Ice version? Why not just a static class called `Version`?

**Comment by pepone** [comment]
Util class has GetProcesLogger/SetProcsesLogger and IntVersion()/StringVersion()

To me putting the version as a static property of the Communicator makes sense because the Communicator is the entry point for the run-time, and these methods return the runtime version, as per the process logger is the same this is the process logger used by Ice run-time.

**Comment by pepone** [comment]
Adding Rutime class sounds fine to me

File: csharp/src/Ice/Runtime.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                    _processLogger = value;
+                }
+            }
+        }
+
+        /// <summary>Returns the Ice version as an integer in the form A.BB.CC, where A indicates the major version,
+        /// BB indicates the minor version, and CC indicates the patch level. For example, for Ice 3.3.1, the returned
+        /// value is 30301.</summary>
+        /// <returns>The Ice version.</returns>
+        public static int IntVersion => 40000; // AABBCC, with AA=major, BB=minor, CC=patch
```
Comment (bernardnormier) [inline csharp/src/Ice/Runtime.cs:38]: Both IntVersion and StringVersion should be const.

File: csharp/src/Ice/Runtime.cs [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+        /// <summary>Returns the Ice version in the form A.B.C, where A indicates the major version, B indicates the
+        /// minor version, and C indicates the patch level.</summary>
+        /// <returns>The Ice version.</returns>
+        public static string StringVersion => "4.0.0-alpha.0"; // "A.B.C", with A=major, B=minor, C=patch
+
+        private static ILogger? _processLogger = null;
```
Comment (bernardnormier) [inline csharp/src/Ice/Runtime.cs:45]: Why not use a volatile field here?

File: csharp/src/Ice/Runtime.cs (hunk 2 above)
Comment (pepone) [reply csharp/src/Ice/Runtime.cs]: Fixed

File: csharp/src/Ice/Runtime.cs (hunk 3 above)
Comment (pepone) [reply csharp/src/Ice/Runtime.cs]: Fixed

**Pull request merged**